File: JS/MaximumRunningTimeOfComputers.test.js
/**
 * @version JAVASCRIPT ECMAScript 6 
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { maxRunTime } from './MaximumRunningTimeOfComputers.js'

TEST('Starting MaximumRunningTimeOfComputers test...', (t) => {
    //Test 1
    let want = 4
    let got = maxRunTime( 2, [3,3,3] )
    t.assert( want.toString === got.toString, "Expect: "+want)
    //Test 2
    want = 2
    got = maxRunTime( 2, [1,1,1,1] )
    t.assert( want.toString === got.toString, "Expect: "+want)

    t.end()
})

File: JAVA/MaximumRunningTimeOfComputersTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class MaximumRunningTimeOfComputersTest {

    @Test
	public void testMaximumRunningTimeOfComputers() throws Exception {
        //Test 1
        long want = 4;
        long got = MaximumRunningTimeOfComputers.maxRunTime( 2, new int[]{3,3,3} );
        assertEquals(want, got, 0);
        //Test 2
        want = 2;
        got = MaximumRunningTimeOfComputers.maxRunTime( 2, new int[]{1,1,1,1} );
        assertEquals(want, got, 0);
	}

}
